geonode/base/api/views.py

```diff
@@ -209,19 +217,26 @@ def resource_types(self, request):
         resource_types = []
         for _model in apps.get_models():
             if _model.__name__ == "ResourceBase":
-                resource_types = [_m.__name__.lower() for _m in _model.__subclasses__()]
-        if "geoapp" in resource_types:
-            resource_types.remove("geoapp")
-        if "service" in resource_types:
-            resource_types.remove("service")
+                resource_types = [{
+                    "name": _m.__name__.lower(),
+                    "count": _m.objects.filter(resource_type=_m.__name__.lower()).count()
+                    }
+                    for _m in _model.__subclasses__()]
+        for name in ['geoapp', 'service']:
+            resource_types = [r_type for r_type in resource_types if r_type['name'] != name]
```

Comment:
what are you doing here? why don't you filter out services before, at the beginning of the for loop?

Reply:
I plan on fixing this after having a common function [here](https://github.com/GeoNode/geonode/pull/7483/files#diff-7a787998e5e7343bea646b19948d0eee47b1f82758b071696fd0767568782b27R147)